src/HarmonicBarrier.h

```diff
@@ -82,19 +113,43 @@ class PYBIND11_EXPORT ImplicitEvaporator : public ForceCompute
      * \param cutoff Distance from potential minimum to cutoff harmonic potential and switch to
      * linear
      */
-    void setParams(unsigned int type, Scalar k, Scalar offset, Scalar g, Scalar cutoff)
+    void setParams(unsigned int type, const param_type& params)
         {
         assert(type < m_pdata->getNTypes());
         ArrayHandle<Scalar4> h_params(m_params, access_location::host, access_mode::readwrite);
-        h_params.data[type] = make_scalar4(k, offset, g, cutoff);
+        h_params.data[type] = make_scalar4(params.k, params.offset, params.g, params.cutoff);
+        }
+
+    void setParamsPython(std::string type_name, pybind11::dict params)
+        {
+        unsigned int type_idx = m_pdata->getTypeByName(type_name);
+        param_type h_params(params);
+        setParams(type_idx, h_params);
+        }
+
+    pybind11::dict getParams(std::string type_name)
```

Comment:
For completeness, I would like to see this as a Python-specific getter, then also include a C++ getter `getParams` that just returns the actual struct (the complementary method to `setParams`). `getParamsPython` can internally use `getParams` before converting it to a dict.
```suggestion
    pybind11::dict getParamsPython(std::string type_name)
```